pubsub/topic.go

```diff
@@ -873,6 +878,83 @@ func (t *Topic) publishMessageBundle(ctx context.Context, bms []*bundledMessage)
 	if err != nil {
 		log.Printf("pubsub: cannot create context with tag in publishMessageBundle: %v", err)
 	}
+
+	// We have to do some weird things to instrument this batcher to get around
+	// OpenTelemetry limitations, namely that you can't add span links after you've created
```

Comment:
This is technically changing at some point soon with `AddLinks` now being stable, go being tracked [here](https://github.com/open-telemetry/opentelemetry-go/issues/3919)

Reply:
Ah yes, I found this after looking around for some time. Very glad it will be changing soon, if a bit too late to help us in this case.